llvm/test/CodeGen/RISCV/rvv/fixed-vectors-shuffle-exact-vlen.ll

```diff
@@ -168,12 +168,11 @@ define <4 x i64> @m2_splat_into_slide_two_source_v2_lo(<4 x i64> %v1, <4 x i64>
 define <4 x i64> @m2_splat_into_slide_two_source(<4 x i64> %v1, <4 x i64> %v2) vscale_range(2,2) {
 ; CHECK-LABEL: m2_splat_into_slide_two_source:
 ; CHECK:       # %bb.0:
-; CHECK-NEXT:    vsetivli zero, 1, e8, mf8, ta, ma
```

Comment:
Did I miss a separate commit which added additional testing for this change?  It clearly isn't NFC, and the test changes here seem fairly minimal for something this involved.  If not, please commit additional coverage (including the cause of the functional bug) before reapplying.

Reply:
You had added a couple tests in 78ab771;  I had missed finding that on my first quick search.

Please add the following:
- A fully quadratic shuffle, possibly at multiple lmuls
- A mostly linear case - i.e. something which *isn't* linear, but you think is still a good example of the generality.
- Whatever the functional bug was.